x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/connectors/connectors_table.tsx

```diff
@@ -181,6 +158,36 @@ export const ConnectorsTable: React.FC<ConnectorsTableProps> = ({
       ),
     },
   ];
+
+  if (!isCrawler) {
```

Comment:
Instead of splice we use spreading as much as possible.
i.e. 
```javascript
const columns = [
...(condition ? [{column: 'data'}]: []),
{
column2: 'data'
},
...(condition ? [{ column3: 'data'}]: []),
]
```

Reply:
So _this_ is how to do it inside an array. Thank you!